### PR TITLE
DENA-974: msk topic: implemented check for the team prefix

### PR DESCRIPTION
**Pull request by sbuliarca**
Within this rule we can add the config checks for it later.

Update: actually, let's keep this separate, as we don't want this to be ignorable

### CODE REVIEW

File: rules/msk_topic.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		return fmt.Errorf("decoding name for kafka_topic '%s': %w", resourceName, diags)
+	}
+
+	if !strings.HasPrefix(topicName, teamName+".") {
+		err := runner.EmitIssue(
+			r,
+			fmt.Sprintf("topic name must have as a prefix the team name '%s'. Current value is '%s'", teamName, topicName),
+			nameAttr.Range,
+		)
+		if err != nil {
```
Comment (MarcinGinszt) [inline rules/msk_topic.go:87]: this condition is always true

Comment (sbuliarca) [reply rules/msk_topic.go:87]: I don't understand. Why would it be always true? The tests do no catch this.

Comment (MarcinGinszt) [reply rules/msk_topic.go:87]: Oh, sorry! My bad :sweat:  you are right

Comment (matthewhughes-uw) [reply rules/msk_topic.go:87]: I'm starting to think we can just ignore this error, glancing at https://github.com/terraform-linters/tflint-ruleset-aws/, a naive search shows:

```
$ git grep 'EmitIssue' -- '*.go' | wc --lines
2717
$ git grep 'EmitIssue.*=' -- '*.go' | wc --lines
5
```

i.e. 2.7k `EmitIssue` calls, only 5 of which are assigned to something (i.e where the error is checked)

Comment (sbuliarca) [reply rules/msk_topic.go:87]: In https://github.com/terraform-linters/tflint-ruleset-terraform/ the error is handled. TBH I don't mind going with the idiomatic go approach of handling it.

**Comment by sbuliarca** [comment]
On a run, seems that bex and iam have topics that do not comply to this.
I'm thinking to add some additional global configuration to allow other prefixes for a team

File: rules/msk_topic_name.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+`team_aliases` maps a team name to it's allowed aliases.
+
+## Example
+
+### Good example
+
+Good for team `pubsub` :
+```hcl
+resource "kafka_topic" "good_topic1" {
+	name = "pubsub.good-topic"
```
Comment (matthewhughes-uw) [inline rules/msk_topic_name.md:28]: pedantic nit: this example uses tabs for indenting, the others use spaces

Comment (sbuliarca) [reply rules/msk_topic_name.md]: Fixed in https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/11/commits/c3117755bfe78eec16ee09b83bf094ab5e4de6dd

File: rules/msk_topic_name.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	if hasTeamNameOrAliasPrefix(topicName, teamName, teamAliases) {
+		return nil
+	}
+
+	var im string
+	if len(teamAliases) != 0 {
+		im = fmt.Sprintf(
+			"topic name must be prefixed with the team name '%s' or one of its aliases '%s'. Current value is '%s'",
+			teamName,
+			strings.Join(teamAliases, ","),
```
[the remark this answers is not in the document]
Comment (sbuliarca) [reply rules/msk_topic_name.go]: Fixed in https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/11/commits/c3117755bfe78eec16ee09b83bf094ab5e4de6dd

File: rules/msk_topic_name.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+resource "kafka_topic" "good_topic2" {
+	name = "alias_pubsub1.good-topic"
+}
```
[the remark this answers is not in the document]
Comment (sbuliarca) [reply rules/msk_topic_name.md]: Fixed in https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/11/commits/c3117755bfe78eec16ee09b83bf094ab5e4de6dd

**Review by matthewhughes-uw — APPROVED** [review]

**Pull request merged**